### PR TITLE
updated 3D noise array function

### PR DESCRIPTION
**Pull request by Julian-Wyatt**
given array coordinate inputs x,y,z, the function returns 3D array of shape (z,y,x) instead of 1D array.

### CODE REVIEW

**Comment by lmas** [comment]
Hi thank you for this. Would you be willing to update it further and match it with #25?
I'm asking you because your PR was smaller of the two.

In short:
- Add docstring and type hints to to the public func noise3array (not _noise3array)
- Add back the old input variable names and use x_i for the loop vars

File: opensimplex/opensimplex.py [hunk 1]
```diff
@@ -107,11 +107,13 @@ def _noise2a(x, y, perm):
...
-    noise = np.zeros(x.size, dtype=np.double)
-    for i in prange(x.size):
-        noise[i] = _noise3(x[i], y[i], z[i], perm, perm_grad_index3)
-    return noise
+def _noise3a(X, Y, Z, perm, perm_grad_index3):
+    noise = np.zeros(X.size * Y.size * Z.size, dtype=np.double)
+    for z in prange(Z.size):
+        for y in prange(Y.size):
+            for x in prange(X.size):
+                noise[(y * Y.size + x)+(z * Y.size * X.size)] = _noise3(X[x], Y[y], Z[z], perm, perm_grad_index3)
```
Comment (lmas) [inline opensimplex/opensimplex.py:115]: Mixing upper/lower case single variables is just confusing and hard to read.
It's better to keep the original inputs (lower case x, y, z) so the function signature doesn't change unnecessarily and instead call the loop vars something else (ie, add some extra letter).

Comment (Julian-Wyatt) [reply opensimplex/opensimplex.py:115]: Sure no worries! 
Shall get onto that now

**Comment by Julian-Wyatt** [comment]
Hope those two commits fix the issues and help. I've used the other PR for consistency.

**Comment by lmas** [comment]
Whoa hold up now mister. I appreciate your time and effort, thanks, but now you've started to add in _partial code_ that's unrelated to your initial PR.

_It's not my intention to sound like an ungrateful muppet, I just want to help you get it right while keeping things clean for everyone else :)_

It's way easier to work with small and clean PRs that contains just a single feature/update/whatever. So your initial PR was good. But it would had been better to open up a new PR for that third commit instead.

**Comment by lmas** [comment]
Can you revert the commit with the new fractal noise func and add it as a separate PR? And please run the new formatter on your code.

**Comment by Julian-Wyatt** [comment]
I've removed the ident from the doc string.
I've reverted the final commit, adding the potential new feature. Instead added this as another PR, building on from this branch.

**Comment by lmas** [comment]
Thank you, looking better now! I don't have time during the week so I'll try to verify and merge this weekend 👍

**Pull request merged**